src/App.js

```diff
@@ -14,13 +16,42 @@ function App() {
         setNumbers(data.board);
       });
   }, []);
-  function renderCellRow(index) {
-    const numbArray = numbers[index].map((number, index) => {
-      return <Cell key={index} value={number} />;
+
+  function renderCellRow(rowNumber) {
+    const numbArray = numbers[rowNumber].map((number, index) => {
+      return (
+        <Cell
+          key={index}
+          value={number}
+          id={[rowNumber, index]}
+          onClick={handleCellClick}
+        />
+      );
     });
     return numbArray;
   }
 
+  function renderNumberBtns() {
+    const numbBtns = [...Array(9)].map((item, index) => (
+      <NumberInput
+        key={index}
+        onClick={handleNumberInputClick}
+        value={index + 1}
+      />
+    ));
+    return numbBtns;
+  }
+
+  function handleCellClick(cellID) {
+    selectedCell = cellID;
+  }
+
+  function handleNumberInputClick(value) {
+    let newNumbArr = [...numbers];
+    newNumbArr[selectedCell[0]][selectedCell[1]] = value;
+    setNumbers(newNumbArr);
+  }
+
```

Comment:
when no cell is selected, the NumberInputCell will be added in the top left cell (probably because it's 0 or ""). maybe you could check first if selectedCell defined and then go ahead with the setting the number?

Reply:
Ah yeah, I mentioned it as a bug and I sort of solved it. (in the next PR) 
But I think I over complicated things. Just checking in the handle click function is much easier 😆

Reply:
haha just saw it! sorrry